server/src/main/java/com/openelements/benchscape/server/store/services/EnvironmentService.java

```diff
@@ -98,6 +89,65 @@ protected EnvironmentEntity createNewEntity() {
         return new EnvironmentEntity();
     }
 
+    public List<Environment> getFilteredEnvironments(String name, String gitOriginUrl, String gitBranch, String systemArch,
```

Comment:
Please split this method: I the repository (`EnvironmentRepository`) we have a function public `List<EnvironmentEntity> getFilteredEnvironments(String name, String gitOriginUrl,...)` that does mostly all of the given function but do not convert the `EnvironmentEntity` to `Environment`. In this service the given method will than call the method in repo and just do the conversion of the result.